### DISCUSSION TITLE
Showing more data on a relation select field in forms

### DISCUSSION THREAD

**Discussion by mh-keshavarz** [Budibase/budibase#3690] [Ideas]
Hi.
I have a field selecting objects of another table with relation in my form. That table has no column suitable for display column on its own so i was trying to show two column in each row of my select form field. 
I first tried to add multiple display fields but it was impossible.
Then i tried editing what each row of select field displays and didn't find how to do it.
At last i tried to add a disabled text field showing value of my other column next to select field based on what selected on the select field but it seems that {{ Form.Fields.[[Value]] }} is not updated when i choose a value in the Value select field and it shows nothing.

**Comment by mjashanks** [comment]
Hi @mh-keshavarz - assuming that you are using a Budibase Table - create a new formula column, then set it to the primary display column.

**Reply by mh-keshavarz** [reply]
Thanks for your fast reply.
Actually i am using a form like this:
![image](https://user-images.githubusercontent.com/7413714/144831297-b40c7ee7-5b55-42bf-a339-2820df9421cf.png)
here the community relation display field is slug but our users need to also see subject when choosing a community. At the moment I added a searchable table below where they can search with slug and see subject of matching communities. but here the user should type the slug is search field.

**Reply by mjashanks** [reply]
I understand - have you tried adding a Formula column and using it as the primary display?

**Reply by mh-keshavarz** [reply]
Oh! As i didn't used the internal datasource i wasn't familiar with concept of formula columns and thought that it is related to display tables.
I am not using built in datasource (using our production postgresql), can i use something like that in my case?

**Reply by mjashanks** [reply]
Yes - you can add a new formula column to a postgres table.

Click on your table and click the `Create column` button. You can then define a formula in Javascript to Handlebars.

![image](https://user-images.githubusercontent.com/3524181/144836161-c1c0daa9-9de8-4d01-9be0-fcd3489f1afc.png)

**Reply by mjashanks** [reply]
You could also use this if you'd prefer: https://www.postgresql.org/docs/12/ddl-generated-columns.html
